### PR TITLE
MNT: Deprecate annexstatus

### PR DESCRIPTION
**Pull request by adswa**
This PR works towards #6076, which calls out the ``AnnexRepo``'s ``annexstatus`` method is undocumented and mostly unused with the exception of tests. 

Its remaining usecases are addressed as follows:
- A few spots (in ``save`` and ``GitRepo``) rely on the existence of this method in ``ds.repo`` to assert whether or not we're dealing with an AnnexRepo. These spots were changed to instead check for a ``uuid`` property, which should exist in AnnexRepos but not in GitRepos
- A few tests were using ``annexstatus``. Some were able to change to use ``get_content_annexinfo``. For those that didn't, I moved ``annexstatus`` into ``tests/utils`` as a new test helper ``get_annexstatus``.

#### 🏠 Internal
- AnnexRepo's internal ``annexstatus`` is deprecated. In its place, a new test helper assists the few tests that rely on it.

### CODE REVIEW

**Comment by adswa** [comment]
Travis failure seems unrelated
```
======================================================================

ERROR: datalad.distribution.tests.test_get.test_source_candidate_subdataset
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/dl-miniconda-chiq2wk1/lib/python3.9/site-packages/nose/case.py", line 134, in run
    self.runTest(result)
  File "/tmp/dl-miniconda-chiq2wk1/lib/python3.9/site-packages/nose/case.py", line 152, in runTest
    test(result)
AssertionError: Test was too slow (took 30.1101s, threshold was 30.0000s)
```

and Appveyor failure is a coverage glitch:
```
python -m coverage xml
C:\Python310\python.exe: No module named coverage
Command exited with code 1
```

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -3334,6 +3334,12 @@ def get_content_annexinfo(
         return info
 
     def annexstatus(self, paths=None, untracked='all'):
+        """
+        .. deprecated:: 0.16
+
+        Use get_content_annexinfo() or the test helper
+        datalad/utils/get_annexstatus() instead.
```
Comment (mih) [inline datalad/support/annexrepo.py:3341]:
```suggestion
        .. deprecated:: 0.16
            Use get_content_annexinfo() or the test helper
            datalad/utils/get_annexstatus() instead.
```

**Review by mih — COMMENTED** [review]
LGTM, thx! Small indentation issue, fix suggested.

**Review by mih — APPROVED** [review]
I merged the doc-fix. Once tests passed, this can move in. Thx!

File: datalad/core/local/save.py [hunk 2]
```diff
@@ -309,7 +309,7 @@ def save_ds(args, version_tag=None):
                         # calls
                         paths=None,
                         # prevent whining of GitRepo
-                        git=True if not hasattr(ds.repo, 'annexstatus')
+                        git=True if not hasattr(ds.repo, 'uuid')
```
Comment (bpoldrack) [inline datalad/core/local/save.py:312]: I'd prefer to use that occasion to unify this kind of test:

```suggestion
                        git=True if not isinstance(ds.repo, AnnexRepo)
```

Comment (bpoldrack) [reply datalad/core/local/save.py:312]: Not a blocker, though.

Comment (mih) [reply datalad/core/local/save.py:312]: Let's do this elsewhere.

**Pull request merged**